File: src/main/java/com/github/kuro46/embedscript/GsonHolder.java
package com.github.kuro46.embedscript;

import com.google.gson.Gson;

import java.lang.ref.WeakReference;

/**
 * @author shirokuro
 */
public final class GsonHolder {
    private static WeakReference<Gson> gsonRef = new WeakReference<>(null);

    private GsonHolder() {
    }

    public static Gson get() {
        Gson gson = gsonRef.get();
        if (gson == null) {
            gsonRef = new WeakReference<>(gson = new Gson());
        }
        return gson;
    }
}

File: src/main/kotlin/com/github/kuro46/embedscript/GsonHolder.kt
package com.github.kuro46.embedscript

import com.google.gson.Gson

import java.lang.ref.WeakReference

/**
 * @author shirokuro
 */
object GsonHolder {
    private var gsonRef = WeakReference<Gson>(null)

    fun get(): Gson {
        var gson = gsonRef.get()
        if (gson == null) {
            gson = Gson()
            gsonRef = WeakReference(gson)
        }
        return gson
    }
}
